empress/tools.py

```diff
@@ -278,7 +277,7 @@ def match_inputs(
     #
     # All that's left to do is to filter the sample metadata to just the
     # samples that are also present in the table.
-    sf_sample_metadata = sample_metadata.loc[ff_table.columns]
+    sf_sample_metadata = sample_metadata.loc[ff_table.ids()]
```

Comment:
```suggestion
    sf_sample_metadata = sample_metadata.loc[table_samples]
```

(I think this should work, but I guess it might alter the order of samples in the metadata somewhat. Feel free to ignore this.)

Reply:
Keeping so sample order doesn't change. It's effectively just a get on an attribute so overhead is very low.